backend/rag-pipeline/llm_interaction.py

```diff
@@ -0,0 +1,127 @@
+import os
+from typing import List, Dict, Optional
+from datetime import datetime
+
+from langchain_huggingface import HuggingFaceEmbeddings
+from langchain_ollama import OllamaLLM
+from langchain_community.vectorstores import FAISS
+from langchain.prompts import PromptTemplate
+from langchain_core.documents import Document
+
+from dotenv import load_dotenv
+load_dotenv()
+
+class NigerianLawRAG:
+    
+    def __init__(self, model_type: str = "ollama", model_name: Optional[str] = None):
+        
+        self.faiss_index_path = os.path.join("/app", "data", "faiss_index") 
+        self.model_type = model_type
+        self.model_name = os.getenv("OLLAMA_MODEL_NAME")
+        self.max_context_length = 3500
+        
+        self.embedding_model = HuggingFaceEmbeddings(model_name='sentence-transformers/all-MiniLM-L6-v2')
+        print("Embedding model loaded.")
+        
+        self.llm = self.initialize_llm()
+        
+        self.vector_store = self.load_vector_store()
+        
+        self.prompt_template = PromptTemplate(
+            template="""You are an expert on Nigerian laws. Use the following context to answer the question accurately and informatively.
+            If the context doesn't contain enough information, state clearly that you cannot answer based on the provided information.
+            Always provide specific dates, names, and events when available.
+            Keep your answer informative but concise.
+
+            Context:
+            {context}
```

Comment:
you must check that the right context is being loaded, you should probably log it to confirm